dags/oup/parser.py

```diff
@@ -0,0 +1,36 @@
+import xml.etree.ElementTree as ET
+
+from common.parsing.parser import IParser
+from common.parsing.xml_extractors import CustomExtractor
+from common.utils import get_text_value
+from structlog import get_logger
+
+
+class OUPParser(IParser):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+        self.article_type_mapping = {
+            "research-article": "article",
+            "corrected-article": "article",
+            "original-article": "article",
+            "correction": "corrigendum",
+            "addendum": "addendum",
+            "editorial": "editorial",
+        }
+
+        extractors = [
+            CustomExtractor(
+                destination="dois", extraction_function=self._get_dois, required=True
+            ),
+        ]
+        super().__init__(extractors)
+
+    def _get_dois(self, article: ET.Element):
+        source = ".front/article-meta/article-id/[@pub-id-type='doi']"
+        doi_element = article.find(source)
+        dois = get_text_value(doi_element)
+        if not dois:
+            return []
+        self.logger.msg("Parsing dois for article", dois=dois)
+        self.dois = dois
+        return [dois]
```

Comment:
aren't dois a string? if yes, shouldn't you split? or maybe it's just one doi, then shouldn't you rename var to `doi`?

Reply:
I should split doi? What do you mean?
Actually, you can get here just one doi.
Dois because t keep it consistent with all other publishers

Reply:
1. You do `get_text_value(doi_element)`
2. If there are multiple dois, you get something like `"10.1103/PhysRevLett.7.394, 10.1103/PhysRevLett.8.928"` (cause you get text) so I am just wondering how it is possible that you can return an array without splitting the result of  `get_text_value(doi_element)`

Reply:
there is just one doi by field, it means get_text_value(doi_element) will return just one doi.
One article can have just one doi.  Array - just because to align with other deployed publishers. It has to be fixed later 😅

Reply:
ok then 👍 so maybe rename the variable to `doi` instead of `dois`, then it would be less ambiguous ;)